### DISCUSSION TITLE
How to use LEDS

### DISCUSSION THREAD

**Discussion by jojo22904** [OpenStickFoundation/GP2040-CE#126] [Q&A]
As far as what LEDS I'm using quite honestly I'm not really sure. I was just curious what could be used and how to use them before I bought any. I was just confused on how it worked because it seemed like most information I found people kinda assumed you knew how to use LEDS with the board.

Also as of now my setup is just some push buttons on a breadboard I'm not really sure what buttons to get either.

**Comment by TheTrainGoes** [comment]
Most any momentary switches will do.  They come in all shapes and sizes.

For LEDs, you do need something a bit more specific.  

I like to use WS2812B or SK6812 MINI-E.  Most addressable RGB LED strips you will find on Amazon or AliExpress will be WS2812B and are just fine for this project.

**Reply by jojo22904** [reply]
Okay thanks! So which pins do you connect the leds to? And which part of the led? Like which part of the leds do you connect to which pin if that makes sense

**Comment by TheTrainGoes** [comment]
For the standard GP2040-CE v0.5.2 the RGB LED pin is GP28.

I would recommend giving the Fighting Lights Pico documentation a read over as it has a lot of great into on how to work with addressable RGB LEDs.

https://docs.google.com/document/d/1qY4HESYdRyFT8OaB5leyHsWm4kfqoA74zqSLAdihuvo/edit

Specifically section 3 - The Neopixel LEDs

**Reply by jojo22904** [reply]
Oh thanks that was very helpful! So do I have to use a brook board or is there a way to only use the Pico?

**Reply by TheTrainGoes** [reply]
GP2040-CE has RGB LED control built into it from a Pico.  If you are looking to use it on consoles like xBox or PS4 / PS5 you would need a Brook board and something like the Fighting Lights Pico project.

**Reply by jojo22904** [reply]
So if I just wire up the leds in series and connect them to GP28, 5v and ground that will work?

**Reply by TheTrainGoes** [reply]
It is recommended that you use a logic converter, resistor and a capacitor as well however if you just want to try it out you can skip those

**Reply by jojo22904** [reply]
Okay!! One last question and I swear I'm done bothering you. I know you kinda already answered this but as far as choosing a button I guess there's really no button that works with a particular led you just have to find a clear button and mount the led underneath?

Also do you have any recommendations for buttons? I'm thinking arcade type buttons. 

Oh yeah one more thing. I know GP2040 is supposed to work with Brook adapters but do you know if it would work with mayflash adapters? Or any other adapter you may know of.
